> ../dbms/src/Common/new_delete.cpp:41:20: error: always_inline function might not be inlinable [-Werror=attributes]
 ALWAYS_INLINE void untrackMemory(void * ptr [[maybe_unused]], std::size_t size [[maybe_unused]] = 0) noexcept

--- a/libs/libcommon/include/common/memory.h
+++ b/libs/libcommon/include/common/memory.h
@@ -15,10 +15,26 @@
     #define USE_JEMALLOC 0
     #include <cstdlib>
 #endif
+#else
+#include <cstdlib>
 #endif
 
-#define ALWAYS_INLINE inline __attribute__((__always_inline__))
-#define NO_INLINE __attribute__((__noinline__))
+// Also defined in Core/Defines.h
+#if !defined(ALWAYS_INLINE)
+#if defined(_MSC_VER)
+    #define ALWAYS_INLINE inline __forceinline
+#else
+    #define ALWAYS_INLINE inline __attribute__((__always_inline__))
+#endif
+#endif
+
+#if !defined(NO_INLINE)
+#if defined(_MSC_VER)
+    #define NO_INLINE static __declspec(noinline)
+#else
+    #define NO_INLINE __attribute__((__noinline__))
+#endif
+#endif
 
 namespace Memory
 {

